### ISSUE TITLE
npm@5 installing a subdependency at root level causes unmet dependencies

### ISSUE THREAD

**Issue by eknkc** [npm/npm#16741]
#### I'm opening this issue because:

  - [X] npm is producing an incorrect install.

#### What's going wrong?

When I install a package `A` that depends on `B@2.0.0` and then install `B@1.0.0` seperately, package A is left with no `B` meeting it.

#### How can the CLI team reproduce the problem?

```sh
$ npm install debug
$ npm ls

root
└─┬ debug@2.6.8
  └── ms@2.0.0

$ npm install ms@1.0.0
$ npm ls

root
├─┬ debug@2.6.8
│ └── UNMET DEPENDENCY ms@2.0.0
└── ms@1.0.0

npm ERR! missing: ms@2.0.0, required by debug@2.6.8
```
### supporting information:

 - `npm -v` prints: 5.0.0
 - `node -v` prints: v7.10.0
 - `npm config get registry` prints: https://registry.npmjs.org/
 - Windows, OS X/macOS, or Linux?: macOS

**Comment by schmod** [comment]
@eknkc Are you able to reproduce this with npm@4?  

It sounds very similar to #10727.

**Comment by zkat** [comment]
This has been fixed, probably by #17508 is my guess. The patch is available in canary and will be included in `npm@5.1.0`.
